contracts/utils/TransientReentrancyGuard.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.24;
+
+import {TransientStorage} from "./TransientStorage.sol";
+
+/**
+ * @dev Variant of {ReentrancyGuard} that uses transient storage.
+ *
+ * NOTE: This variant only works on networks where EIP-1153 is available.
+ */
+abstract contract TransientReentrancyGuard {
```

Comment:
What about `ReentrancyGuardTransient` instead? I feel the "transient" prefix captures the show:

```suggestion
abstract contract ReentrancyGuardTransient {
```